src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +88,163 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        this.removeObsoleteData(this.schema, schema, this.data);
+        this.loadAndMergeOriginData(this.schema, schema).then(action(()=> {
+            this.schema = schema;
+            this.addMissingSchemaProperties();
+            this.setSchemaLoading(false);
+        }));
     };
 
+    loadAndMergeOriginData(oldSchema: Schema, newSchema: Schema) {
+        // origin data must be loaded only when switching back to the origin template
```

Comment:
i think im missing something here 🙈  
what is the `origin template` exactly? why dont we need to merge the data for other templates?

Reply:
the `originTemplate` is the template which is saved in the backend. We only need to merge the data, when the user goes back to the `originTemplate` because then we can load the `originData` for this template.

If the user switches between other templates, we just remove the invalid properties for the new schema, but we have no other `data` to make a merge.